File: web-api/src/main/java/pl/sgorski/nethelt/webapi/features/agent/dto/response/AgentResponse.java
package pl.sgorski.nethelt.webapi.features.agent.dto.response;

import java.time.Instant;
import org.jspecify.annotations.Nullable;
import pl.sgorski.nethelt.webapi.features.agent.domain.AgentStatus;

public record AgentResponse(
    Long id,
    String name,
    AgentStatus status,
    @Nullable Instant lastHeartbeatAt,
    Instant tokenCreatedAt) {}

File: frontend/src/app/models/agent/agent-response.ts
import { AgentStatus } from './agent-status';

export interface AgentResponse {
  id: number;
  name: string;
  status: AgentStatus;
  lastHeartbeatAt: string | null;
  tokenCreatedAt: string;
}

export interface AgentTokenResponse {
  token: string;
}
